### PR TITLE
extend: Add instructions for manual plugin SDK migration

### PR DESCRIPTION
**Pull request by radeksimko**

### CODE REVIEW

**Review by appilon — APPROVED** [review]

File: content/source/docs/extend/plugin-sdk.html.md [hunk 1]
```diff
@@ -82,11 +84,74 @@ $ go mod vendor
...
+   - `go version` should yield the current version in use
+ - Dependencies managed via Go modules
+   - `go.mod` and `go.sum` should be present in the root
+ - `github.com/hashicorp/terraform` `>=0.12.7`
+   - `go.mod` should contain a line which starts with `github.com/hashicorp/terraform` and version
+     that is greater than or equal to `0.12.7`
+
+#### Step 1: Check for deprecations
+
+See the [list of deprecations](#deprecations) below and take actions to remove all occurences
```
Comment (kmoe) [inline content/source/docs/extend/plugin-sdk.html.md]:
```suggestion
See the [list of deprecations](#deprecations) below and take actions to remove all occurrences
```

File: content/source/docs/extend/plugin-sdk.html.md [hunk 2]
```diff
@@ -82,11 +84,74 @@ $ go mod vendor
...
+of deprecated packages, functions or identifiers.
+
+You may find full list of SDK packages in [`tf-sdk-migrator` source code](https://github.com/hashicorp/tf-sdk-migrator/blob/c7297e03e62319a4eb48f4bc2fd1d8ee91ecade0/cmd/check/sdk_imports.go#L8-L21). Any package which is not on the list is considered as deprecated
```
Comment (kmoe) [inline content/source/docs/extend/plugin-sdk.html.md]:
```suggestion
You may find a full list of SDK packages in [`tf-sdk-migrator` source code](https://github.com/hashicorp/tf-sdk-migrator/blob/c7297e03e62319a4eb48f4bc2fd1d8ee91ecade0/cmd/check/sdk_imports.go#L8-L21). Any package which is not on the list is considered as deprecated
```

File: content/source/docs/extend/plugin-sdk.html.md [hunk 3]
```diff
@@ -82,11 +84,74 @@ $ go mod vendor
...
+in the context of SDK and/or doesn't classify as SDK.
+
+You can use standard Go tooling, [JQ](https://stedolan.github.io/jq/) and [grep](https://en.wikipedia.org/wiki/Grep)
```
Comment (kmoe) [inline content/source/docs/extend/plugin-sdk.html.md]:
```suggestion
You can use standard Go tooling, [jq](https://stedolan.github.io/jq/) and [grep](https://en.wikipedia.org/wiki/Grep)
```

File: content/source/docs/extend/plugin-sdk.html.md [hunk 4]
```diff
@@ -82,11 +84,74 @@ $ go mod vendor
...
+go list -json ./... | \
+	jq -r .Imports[] | \
+	grep '^github.com/hashicorp/terraform/'
+```
+
+You can use [`go-refs`](https://github.com/radeksimko/go-refs) to list all identifiers of a given package which are in use.
+
+#### Step 2: Replace import paths
+
+The simplest way to replace all import paths is to find all Go files using the standard Go tooling and JQ,
```
Comment (kmoe) [inline content/source/docs/extend/plugin-sdk.html.md]:
```suggestion
The simplest way to replace all import paths is to find all Go files using the standard Go tooling and jq,
```

**Review by kmoe — APPROVED** [review]
Great docs, LGTM @radeksimko. Some optional proofreading suggestions.

**Pull request merged**